### ISSUE TITLE
bug: Stencil dev server CSS updates are broken when using `dist-hydrate-script`

### ISSUE THREAD

**Issue by alicewriteswrongs** [ionic-team/stencil#3461]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.17.0

### Current Behavior

**summary**: if `dist-hydrate-script` output target is set then CSS hot-reload is broken and changing a CSS file will result in components losing all of their styling.

This problem has to do with how we rewrite CSS in order to support scoped components, an alternative way to scope styles to a component which does not use the shadow DOM. We have a utility function, defined in `src/utils/shadow-css.ts`, which rewrites a CSS string in order to 'scope' it within a component by rewriting the selectors to include a unique ID. This function is called in the dev server workflow in such a way that CSS gets mangled if (possibly iff):

- a component has `shadow: true` and
- the `dist-hydrate-script` output target is set in the Stencil configuration

This results in styles targeting the `:host` selector to be rewritten to a different selector, wiping styles from the component when the updated CSS is applied to the page in the browser.

### Why is this happening?

The `scopeCss` function is defined here:

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/utils/shadow-css.ts#L418

This function is called when the dev server is processing a change event to a CSS file. The function is called in the `transformCssToEsmModule` function in `src/compiler/style/css-to-esm.ts`:

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/compiler/style/css-to-esm.ts#L115-L120

The third argument to the function is here set to `input.commentOriginalSelector`, which causes the CSS selectors to be commented out (see below). This causes CSS using the `:host` selector for styling the component's shadow DOM to be mangled in such a way that the styles don't apply to anything.

The third argument is set in the `extTransformsPlugin`, here:

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/compiler/bundle/ext-transforms-plugin.ts#L54-L65

and here:

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/compiler/bundle/ext-transforms-plugin.ts#L34

`bundleOpts.platform` is in turn set to `"hydrate"` in the `BundleOptions` object constructed in `bundleHydrateFactory`, here:

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/compiler/output-targets/dist-hydrate-script/bundle-hydrate-factory.ts#L19-L31

which is in turn called in the `generateHydrateApp` function which does the main work for the `dist-hydrate-script` output target:

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/compiler/output-targets/dist-hydrate-script/generate-hydrate-app.ts#L83

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/compiler/output-targets/dist-hydrate-script/index.ts#L5-L14

Anyhow, to make a long story short, when the `dist-hydrate-script` output target is set then CSS will be 'scoped' using the `scopeCss` function with the `commentOriginalSelector` argument set to `true`, even in situations in which it isn't appropriate to do so, and in particular, when a component does _not_ have `scoped: true` and instead has `shadow: true`. This will cause CSS like:

```css
:host {
    background: blue;
}
```

to be rewritten like

```css
/*!@:host*/.my-component-h{background: blue;}
```

This, in turn, results in the CSS not applying to the component, and we get a nice, unstyled component instead of what we want.

### Why doesn't this happen on the first page load?

I believe this doesn't happen on first page load because on first page load CSS is processed in the `initializeComponent` function but in that instance it is not called with the `commentOriginalSelector` argument set to `true`:

https://github.com/ionic-team/stencil/blob/658fdd86d5fde804cfe318317c6276b287b4fcbe/src/runtime/initialize-component.ts#L104-L111

### Expected Behavior

The hot module reload functionality in the dev server should work, i.e. edits made in the CSS file for a component should be reflected in the browser automatically without having to reload the page.

### Steps to Reproduce

There are detailed instructions in the reproduction repo.

### Code Reproduction URL

https://github.com/alicewriteswrongs/stencil-hmr-hydrate-repro

**Comment by alicewriteswrongs** [comment]
The offending line (the `encapsulation === "shadow"` check) was added in this commit in 2020: https://github.com/ionic-team/stencil/commit/b5757df7c0737a3006668160da4d346047defd06

https://github.com/ionic-team/stencil/blob/b5757df7c0737a3006668160da4d346047defd06/src/compiler_next/bundle/ext-transforms-plugin.ts#L26

afaict it isn't necessary? there is no context in the original change other than the commit message "only comment original selectors for shadow css."

**Comment by rwaskiewicz** [comment]
The PR that fixes this issue (#3517) is a part of [today's v2.17.4 release](https://github.com/ionic-team/stencil/releases/tag/v2.17.4). As a result, I'm going to close this issue. Should the issue reoccur on v2.17.4+ of Stencil, please open a new ticket, thanks!
